components/layout_2020/flow/mod.rs

```diff
@@ -143,21 +145,91 @@ impl BlockContainer {
         writing_mode: WritingMode,
     ) -> ContentSizes {
         match &self {
-            Self::BlockLevelBoxes(boxes) if layout_context.use_rayon => boxes
-                .par_iter()
-                .map(|box_| {
-                    box_.borrow_mut()
-                        .inline_content_sizes(layout_context, writing_mode)
-                })
-                .reduce(ContentSizes::zero, ContentSizes::max),
-            Self::BlockLevelBoxes(boxes) => boxes
-                .iter()
-                .map(|box_| {
-                    box_.borrow_mut()
-                        .inline_content_sizes(layout_context, writing_mode)
-                })
-                .reduce(ContentSizes::max)
-                .unwrap_or_else(ContentSizes::zero),
+            Self::BlockLevelBoxes(boxes) => {
+                let get_box_info = |box_: &ArcRefCell<BlockLevelBox>| {
+                    let size = box_
+                        .borrow_mut()
+                        .inline_content_sizes(layout_context, writing_mode);
+                    if let BlockLevelBox::OutOfFlowFloatBox(ref float_box) = *box_.borrow_mut() {
+                        let style_box = match &float_box.contents {
```

Comment:
I think this could be turned into a function `IndependentFormattingContext::style()` which returns a reference to the style.

Reply:
Actually `IndependentFormattingContext::style()` already existed :sweat_smile: